pkg/agentctl/sync.go

```diff
@@ -25,22 +25,22 @@ import (
 // ConfigSync will completely overwrite the set of active configs
 // present in the provided PrometheusClient - configs present in the
 // API but not in the directory will be deleted.
-func ConfigSync(logger log.Logger, cli client.PrometheusClient, dir string) error {
+func ConfigSync(logger log.Logger, cli client.PrometheusClient, dir string, dryRun bool) error {
 	if logger == nil {
 		logger = log.NewNopLogger()
 	}
 
 	ctx := context.Background()
-	existing, err := cli.ListConfigs(ctx)
-	if err != nil {
-		return fmt.Errorf("could not list configs: %w", err)
-	}
-
 	cfgs, err := ConfigsFromDirectory(dir)
 	if err != nil {
 		return err
 	}
 
+	if dryRun {
+		fmt.Println("Dry run successful")
```

Comment:
This should probably use the logger we're passing in:

```
level.Info(logger).Log("msg", "dry run sucessful")
```

OTOH, maybe it's better to say something like "configs have been validated successfully" here? eg something that tells the user what was successful during the dry run in particular.